Harden Dependabot: switch to corepack and restrict TypeScript to v5

## Summary

Two related Dependabot hardening changes:

1. Replace `pnpm/action-setup@v5` with corepack in the CI workflow so Dependabot bumps to `pnpm/action-setup@v6` can be ignored safely.
2. Add an explicit `ignore` rule for TypeScript major version updates in `.github/dependabot.yml`.

---

## 1. Switch CI from `pnpm/action-setup` to corepack

### Why

`pnpm/action-setup@v6` installs pnpm via npm, and the npm-installed pnpm bundles a YAML parser that rejects `pnpm-lock.yaml` v9.0 format. Additionally, `actions/setup-node@v6` with `cache: pnpm` rewrites the lockfile when it detects npm-installed pnpm. These two issues combine to fail CI with:

```
ERR_PNPM_BROKEN_LOCKFILE  The lockfile is broken: expected a single document in the stream, but found more
```

A simple bump to v6 does not fix this, so switch to corepack instead. Corepack reads the `packageManager` field from `package.json` and installs the correct standalone binary.

### Changes

Replace all `pnpm/action-setup@v5` usages in `.github/workflows/ci.yml` (currently on lines 24 and 49) with corepack:

```yaml
# Before
- uses: pnpm/action-setup@v5

# After
- name: Enable corepack
  run: corepack enable pnpm
```

### Reference

Discovered and fixed in aicers/github-dashboard#445. Same fix applied in aicers/aimer-web#149.

---

## 2. Restrict Dependabot TypeScript updates to v5

### Why

We are not ready to adopt TypeScript 6 yet (`package.json` currently pins `typescript": "^5"`). Without an explicit `ignore` rule, Dependabot will keep opening PRs to bump to a major version we intend to stay off of.

### Changes

Add an `ignore` rule for `typescript` in `.github/dependabot.yml`, mirroring the existing rules for `node` and `@types/node`:

```yaml
- dependency-name: typescript
  update-types:
    - version-update:semver-major
  versions:
    - ">= 6"
```

### Reference

Same fix applied in aicers/aimer-web#157.

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -23,6 +23,11 @@ updates:
           - version-update:semver-major
         versions:
           - ">= 25"
+      - dependency-name: typescript
+        update-types:
+          - version-update:semver-major
+        versions:
+          - ">= 6"
 
   - package-ecosystem: github-actions
     directory: /

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -21,7 +21,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v6
-      - uses: pnpm/action-setup@v5
+      - name: Enable corepack
+        run: corepack enable pnpm
       - uses: actions/setup-node@v6
         with:
           node-version: 24
@@ -46,7 +47,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v6
-      - uses: pnpm/action-setup@v5
+      - name: Enable corepack
+        run: corepack enable pnpm
       - uses: actions/setup-node@v6
         with:
           node-version: 24

